### DISCUSSION TITLE
view only permissions for tenant?

### DISCUSSION THREAD

**Discussion by sandermanneke** [filamentphp/filament#8730] [Help]
I would like my users to be able to view a Tenant model without being able to edit. 

When I remove the "update" permission and allow "view" and/or "viewAny" the Tenant profile link is removed from the menu and the profile becomes inaccessibel which is unexpected for me.

1) It seems only EditTenantProfile is implemented instead of a more generic ManageTenantProfile. I would expect the profile page the behave as a normal Filament resource with a disabled form when canView is true.

2) I found vendor/filament/filament/src/Pages/Tenancy/EditTenantProfile.php - here I would expect canView to respect authorize('**view**', $tenant) instead of the implemented authorize('**update**', $tenant)

How would I best go about being able to only view a tenant profile based on the permissions?

Or would you consider this a bug / incomplete feature and should it be entered as such?

**Comment by VincvanAs** [comment]
It makes sense that the EditTenantProfile uses the update policy to determine if a user can access and edit the page.

The "issue" here is that you've defined the EditTenantProfile page in your Panel Provider as tenantProfile(). In an ideal scenario, you could define the tenantProfile and tenantProfileEdit pages as separate entities, but that's not the case in the current Filament version.

In your situation, I would suggest adding ->disabled(fn (): bool => !auth()->user()->can('edit-tenant')) and/or hidden to the form fields. This way, users can view the profile page but won't be able to edit it.
